src/main/java/org/ladocuploader/app/interceptors/DataRequiredInterceptor.java

```diff
@@ -6,79 +6,134 @@
 import jakarta.servlet.http.HttpServletResponse;
 import jakarta.servlet.http.HttpSession;
 import lombok.extern.slf4j.Slf4j;
+import org.ladocuploader.app.data.RecoverySubmissionIDs;
 import org.springframework.stereotype.Component;
 import org.springframework.util.AntPathMatcher;
 import org.springframework.web.servlet.HandlerInterceptor;
 
+import java.io.IOException;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.UUID;
 
 @Component
 @Slf4j
 public class DataRequiredInterceptor implements HandlerInterceptor {
-    public static final String PATH_FORMAT = "/flow/{flow}/{screen}";
-    private final SubmissionRepositoryService submissionRepositoryService;
-    private static final Map<String, String> REQUIRED_DATA = Map.ofEntries(
-            // Step 1
-            Map.entry("howToAddDocuments", "firstName"),
-            Map.entry("uploadDocuments", "firstName"),
-            Map.entry("docSubmitConfirmation", "firstName"),
-            Map.entry("finalConfirmation", "firstName")
-    );
-
-    public DataRequiredInterceptor(SubmissionRepositoryService submissionRepositoryService) {
-        this.submissionRepositoryService = submissionRepositoryService;
-    }
+  public static final String PATH_FORMAT = "/flow/{flow}/{screen}";
+  private final SubmissionRepositoryService submissionRepositoryService;
+  private final RecoverySubmissionIDs recoveryRecoverySubmissionIDs;
+
+  // Checks at signposts
+  private static final Map<String, String> REQUIRED_DATA = Map.of(
+      "signPost", "parish",
+      "householdSignPost", "parish",
```

Comment:
I think I'm a little uncomfortable with only doing data checks on certain pages. If I mistakenly jump into the flow somewhere, start entering data, then hit one of these pages, I will bet bounced back and never understand what happened.

Reply:
If the session split still happens the user experience here will be interesting.